src/main/java/com/google/sps/data/FirebaseSingletonApp.java

```diff
@@ -0,0 +1,44 @@
+package com.google.sps.data;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+
+/**
+ * Singleton class that will return the unique Firebase instance within the
+ * entire app.
+ */
+public class FirebaseSingletonApp {
+  private static FirebaseApp instance = initialize();
+
+  /**
+   * Initialize firebase SDK and return the FirebaseApp object
+   */
+
+  private static FirebaseApp initialize() {
+    try {
+      // System.out.println(System.getProperty("user.dir"));
+      FileInputStream serviceAccount = new FileInputStream(System.getProperty("user.home") + 
+          "/step235-2020/serviceAccountKey.json");
+
+      FirebaseOptions options = new FirebaseOptions.Builder()
+          .setCredentials(GoogleCredentials.fromStream(serviceAccount))
+          .setDatabaseUrl("https://alphabooks-step-2020.firebaseio.com").build();
+
+      return FirebaseApp.initializeApp(options, "AlphaBooks");
+    } catch (FileNotFoundException e) {
```

Comment:
Don't catch exceptions you don't know how to handle. Just let them through.

Yes, the logs are huge and it's pain to dig through them. But fighting java on this is pointless. Exceptions and long stack traces are simply something that you should get used to.

Reply:
When I try to throw it in the signature of the initialize() function I get an error at 
private static FirebaseApp instance = initialize();
and I was not sure how else I can solve this.